### DISCUSSION TITLE
Switching from and to Full Screen apps reset the brightness momentarily

### DISCUSSION THREAD

**Discussion by gr4v1ty69** [MonitorControl/MonitorControl#1628] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Looked through [the wiki](https://github.com/MonitorControl/MonitorControl/wiki)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the bug

Switching from normal desktop to a full screen apps resets the brightness to the monitor-set level. Mine is set at 100% while MonitorControl is set at 80%. For a brief second when entering and exiting the full screen app, the brightness reverts to 100%. 

When talking about FS apps, I'm talking about this : 
![image](https://github.com/user-attachments/assets/e7a8b2a7-704c-4a1f-ae8e-b4cb97129d15)

Here's a video of the bug : 
https://github.com/user-attachments/assets/67ede441-c4aa-496d-bd59-cf28ff09986b

### Steps to reproduce

1. Put app in full screen
2. 3 finger swipe between apps

### Expected behavior

Keep same brightness.

### Environment Information (please complete the following information)

```markdown
- macOS version: 15.0
- Mac model:13-inch, 2020, Four Thunderbolt 3 ports
- MonitorControl version: 4.2.0
- Monitor(s): Asus VA24DQ
- Apple Silicon/M1 (yes or no): no
```

**Comment by waydabber** [comment]
This is normal when the app is reverting to overlay based dimming (for displays where neither hardware control nor color table dimming is available) as overlays do not work during full screen transitions.

**Reply by gr4v1ty69** [reply]
I'm not really technical but shouldn't a monitor with the DDC/CI function have hardware control dimming?
